### ISSUE TITLE
Improved AWS Personalize support

### ISSUE THREAD

**Issue by GeorgeBellTMH** [aws-amplify/amplify-cli#5823]
Right now it appears there is partial support for AWS Personalize. 

Missing functionality
--
- cli setup and configuration
- permissions for cognito users (ie unauth, auth etc)
- auto configuration with aws-exports.js
- ability to version control the schema
- ability to get personalized suggestions from amplify API

Existing functionality
--
- AmazonPersonalizeProvider
- ability to use analytics.record to save to personalized items using the AmazonPersonalizeProvider

**Comment by nikhname** [comment]
Hi @GeorgeBellTMH out of curiosity, what is your use case for the missing functionality?

**Comment by GeorgeBellTMH** [comment]
Our use case would be multiple environments with personalized data...right now you have to manually setup each dataset, which obviously is problematic with multiple environments.

The secondary use case would be to make use of the personalized data in our app (which will of course require a way to get the data easily from the service). I realize some of this is CLI (setup and config) and some of it is JS (getting suggestions) related...

**Comment by nipoonp** [comment]
I agree, "ability to get personalized suggestions from amplify API" is an important part, but missing.

**Comment by pcolmer** [comment]
In the absence of the CLI doing **any** sort of setup for Personalize, can we at least have some documentation please that shows what IAM permissions are required for things like video tracking? All we have at the moment is https://docs.amplify.aws/lib/analytics/personalize/q/platform/js/ which is small snippets of code with hardly any other information like how to set up Personalize manually. I've guessed at things like the schema settings but since we're hitting permission problems at the moment, who knows what we've got to do to get this working :(
